controllers/networkfence_controller.go

```diff
@@ -18,19 +18,46 @@ package controllers
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
+	"github.com/go-logr/logr"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
+	conn "github.com/csi-addons/kubernetes-csi-addons/internal/connection"
+	"github.com/csi-addons/kubernetes-csi-addons/internal/proto"
+	"github.com/csi-addons/kubernetes-csi-addons/internal/util"
 )
 
-// NetworkFenceReconciler reconciles a NetworkFence object
+// NetworkFenceReconciler reconciles a NetworkFence object.
 type NetworkFenceReconciler struct {
 	client.Client
+	// Scheme defines methods for serializing and deserializing API objects.
 	Scheme *runtime.Scheme
+	// ConnectionPool consists of map of Connection objects
+	Connpool *conn.ConnectionPool
+}
+
+const (
+	networkFenceFinalizer = "csiaddons.openshift.io/network-fence"
+)
+
+// validateNetworkFenceSpec validates the NetworkFence spec and checks if values are neither nil nor empty.
+func validateNetworkFenceSpec(nwFence *csiaddonsv1alpha1.NetworkFence) error {
```

Comment:
if you really want to validate everything (you should, probably), add checks for:

- `nwFence == nil`
- `nwFence.Spec` == nil`

Reply:
Added now :+1:  

(`nwFence.Spec` cannot be checked directly as it is a struct)